### PR TITLE
Openshift deployment support

### PR DESCRIPTION
**Pull request by ppatierno**
This is a first PR to review and discuss for generating artifacts for a Kubernetes and Openshift deploy.
I'd like to start the discussion around that.

Some missing things are : 
- renaming "config" module that now has only certificates
- trying to use only the fabric8 maven plugin (used for generating Kubernetes/Openshift yaml files) because it contains the docker maven plugin that we already use for building Docker images
- generating different certificates for Kubernetes/Openshift deploy (for now the hostname verification is disabled in order to work)
- updating documentation and web site
- adding health checks on deployments
- creates OpenShift routes for reaching from outside

I'm sure other things will come but I'd like to start the review and merge this branch :-)

### CODE REVIEW

File: application/pom.xml [hunk 1]
```diff
@@ -86,6 +108,8 @@
                     <cmd>
                       <exec>
                         <arg>java</arg>
+                        <arg>-Dvertx.disableFileCaching=true</arg>
```
Comment (sophokles73) [inline application/pom.xml]: why is this necessary?

File: application/pom.xml [hunk 2]
```diff
@@ -86,6 +108,8 @@
...
+                        <arg>-Dvertx.disableFileCPResolving=true</arg>
```
Comment (sophokles73) [inline application/pom.xml]: why is this necessary?

File: application/pom.xml [hunk 3]
```diff
@@ -96,8 +120,38 @@
...
+                      <descriptorRef>artifact</descriptorRef>-->
+                      <mode>dir</mode>
+                      <basedir>/</basedir>
+                      <inline>
+                        <fileSets>
+                          <fileSet>
+                            <directory>${project.basedir}/target/classes/certs</directory>
+                            <outputDirectory>etc/hono/certs</outputDirectory>
+                            <includes>
+                              <include>*.p12</include>
```
Comment (sophokles73) [inline application/pom.xml]: you only need honoKeyStore.p12 for Hono Server

File: application/pom.xml [hunk 4]
```diff
@@ -96,8 +120,38 @@
...
+                              <include>qdrouter-*.pem</include>
```
Comment (sophokles73) [inline application/pom.xml]: You don't need the qdrouter keys for Hono server

File: application/pom.xml [hunk 5]
```diff
@@ -96,8 +120,38 @@
...
+                              <include>trusted-certs.pem</include>
+                              <include>trustStore.jks</include>
```
Comment (sophokles73) [inline application/pom.xml]: contains the same as trusted-certs.pem

File: config/pom.xml [hunk 6]
```diff
@@ -11,90 +11,30 @@
...
   <description>Default Hono configuration to be used by other modules.</description>
 
+  <build>
+    <resources>
+      <resource>
+        <directory>demo-certs/certs</directory>
+        <!--<filtering>true</filtering>-->
+      </resource>
+    </resources>
+    <!--<plugins>
```
Comment (sophokles73) [inline config/pom.xml]: remove commented out sections or explain why the section is commented out (and not removed)

File: dispatchrouter/pom.xml [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                            <includes>
+                              <include>qdrouterd.sasldb</include>
+                              <include>qdrouter-sasl.conf</include>
+                            </includes>
+                          </fileSet>
+                          <fileSet>
+                            <directory>${project.basedir}/target/classes/certs</directory>
+                            <outputDirectory>certs</outputDirectory>
+                            <includes>
+                              <include>*.p12</include>
```
Comment (sophokles73) [inline dispatchrouter/pom.xml]: not needed by dispatch router

File: dispatchrouter/pom.xml [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                              <include>qdrouter-*.pem</include>
+                              <include>trusted-certs.pem</include>
+                              <include>trustStore.jks</include>
```
Comment (sophokles73) [inline dispatchrouter/pom.xml]: not needed by dispatch router

File: server/src/test/resources/permissions.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+      "hono-client": [ "WRITE" ],
+      "rest-adapter": [ "WRITE" ],
+      "mqtt-adapter": [ "WRITE" ]
+  },
+  "registration/DEFAULT_TENANT": {
+      "hono-client": [ "READ", "WRITE" ],
+      "rest-adapter": [ "READ", "WRITE" ],
+      "mqtt-adapter": [ "READ" ]
+  }
+}
```
Comment (sophokles73) [inline server/src/test/resources/permissions.json]: missing new line

File: .travis.yml [hunk 10]
```diff
@@ -5,4 +5,9 @@ jdk:
...
 sudo: required
 services:
-  - docker
\ No newline at end of file
+  - docker
+before_install:
+  - wget https://archive.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.zip
+  - unzip -qq apache-maven-3.3.9-bin.zip
+  - export M2_HOME=$PWD/apache-maven-3.3.9
+  - export PATH=$M2_HOME/bin:$PATH
```
Comment (sophokles73) [inline .travis.yml]: newline?

File: adapters/mqtt-vertx/pom.xml [hunk 11]
```diff
@@ -110,8 +136,28 @@
...
+                      <descriptorRef>artifact</descriptorRef>-->
+                      <mode>dir</mode>
+                      <basedir>/</basedir>
+                      <inline>
+                        <fileSets>
+                          <fileSet>
+                            <directory>${project.basedir}/target/classes/certs</directory>
+                            <outputDirectory>etc/hono/certs</outputDirectory>
+                            <includes>
+                              <include>trustStore.jks</include>
```
Comment (sophokles73) [inline adapters/mqtt-vertx/pom.xml]: I would use trusted-certs.pem instead (doesn't require password)

File: adapters/mqtt-vertx/pom.xml [hunk 12]
```diff
@@ -126,6 +172,22 @@
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>io.fabric8</groupId>
+            <artifactId>fabric8-maven-plugin</artifactId>
```
Comment (sophokles73) [inline adapters/mqtt-vertx/pom.xml]: this plugin seems to expect yaml files being present in `${project.basedir}/src/main/fabric8`, right?

File: config/pom.xml (hunk 6 above)
Comment (ppatierno) [inline config/pom.xml]: Here I have a doubt because we have two solutions. Due to the presence of binary files (p12 and jks), with the initial configuration which had filtering=true I had problems with corrupted binary files by the maven resources plugin. One way is to set filtering=false and the other one is leaving filtering=true but adding the non filtered file extensions (p12 and jks). What do you think ?

File: adapters/mqtt-vertx/pom.xml (hunk 12 above)
Comment (ppatierno) [inline adapters/mqtt-vertx/pom.xml]: Yes

File: adapters/mqtt-vertx/pom.xml (hunk 11 above)
Comment (ppatierno) [inline adapters/mqtt-vertx/pom.xml]: Agree. The current docker compose file is using the jks file for this reason I used that. Can I go ahead and changing here and the docker compose file as well ?

File: application/pom.xml (hunk 2 above)
Comment (ppatierno) [inline application/pom.xml]: Not a big expert on that but file caching and CPResolving seems to be related to cache resources file for fat jar and used a lot by the Vert.x web component. The problem with OpenShift is that they try to create a local folder for caching and it's something that can't be done on OpenShift because the container runs with a user without more privileges. An application running on OpenShift can write on an external volume (i.e. I set up one for the device registry file). Btw Clement from the Vert.x team suggested me to set vertx.CacheDirBase=/tmp where /tmp is another folder available for writing (without the need of a volume).

File: config/pom.xml (hunk 6 above)
Comment (sophokles73) [inline config/pom.xml]: If the only resource we are left with are the certs then I do not see why we would need to filter them.

File: adapters/mqtt-vertx/pom.xml (hunk 11 above)
Comment (sophokles73) [inline adapters/mqtt-vertx/pom.xml]: Sure

File: adapters/rest-vertx/src/main/fabric8/hono-adapter-rest-vertx-pvc.yaml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ${project.artifactId}-pv-claim
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 5Gi
```
Comment (sophokles73) [inline adapters/rest-vertx/src/main/fabric8/hono-adapter-rest-vertx-pvc.yaml]: newline

File: adapters/rest-vertx/src/main/fabric8/hono-adapter-rest-vertx-svc.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+    name: ${project.artifactId}
+  name: ${project.artifactId}
+spec:
+  ports:
+  - name: http
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    name: ${project.artifactId}
```
Comment (sophokles73) [inline adapters/rest-vertx/src/main/fabric8/hono-adapter-rest-vertx-svc.yaml]: newline

File: application/src/main/fabric8/hono-app-pv.yaml [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: ${project.artifactId}-pv
+spec:
+  capacity:
+    storage: 5Gi
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Recycle
+  hostPath:
+    path: /tmp/hono
```
Comment (sophokles73) [inline application/src/main/fabric8/hono-app-pv.yaml]: newline

File: application/src/main/fabric8/hono-app-pvc.yaml [hunk 16]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ${project.artifactId}-pv-claim
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 5Gi
```
Comment (sophokles73) [inline application/src/main/fabric8/hono-app-pvc.yaml]: newline

File: application/src/main/fabric8/hono-app-svc.yaml [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+    name: ${project.artifactId}
+  name: ${project.artifactId}
+spec:
+  ports:
+  - name: amqp
+    port: 5672
+    protocol: TCP
+    targetPort: 5672
+  selector:
+    name: ${project.artifactId}
```
Comment (sophokles73) [inline application/src/main/fabric8/hono-app-svc.yaml]: newline

File: application/src/main/resources/application.yml [hunk 18]
```diff
@@ -1,17 +1,3 @@
...
-  downstream:
-    host: localhost
-    port: 5672
-    name: hono-server
-  maxInstances: 0
-  singleTenant: false
-  server:
-    bindaddress: 127.0.0.1
-    port: 5672
+    active: default
```
Comment (sophokles73) [inline application/src/main/resources/application.yml]: newline

File: dispatchrouter/pom.xml [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+
+</project>
```
Comment (sophokles73) [inline dispatchrouter/pom.xml]: newline

File: dispatchrouter/src/main/fabric8/dispatch-router-svc.yaml [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+  - name: amqp
+    port: 15672
+    protocol: TCP
+    targetPort: 5672
+  - name: internal
+    port: 5673
+    protocol: TCP
+    targetPort: 5673
+  selector:
+    capability: router
```
Comment (sophokles73) [inline dispatchrouter/src/main/fabric8/dispatch-router-svc.yaml]: newline

File: config/pom.xml (hunk 6 above)
Comment (ppatierno) [inline config/pom.xml]: Absolutely agree on that

File: adapters/mqtt-vertx/pom.xml (hunk 11 above)
Comment (ppatierno) [inline adapters/mqtt-vertx/pom.xml]: I'll do ...

File: adapters/mqtt-vertx/pom.xml [hunk 21]
```diff
@@ -100,6 +124,7 @@
                     <cmd>
                       <exec>
                         <arg>java</arg>
+                        <arg>-Dvertx.cacheDirBase=/tmp</arg>
```
Comment (sophokles73) [inline adapters/mqtt-vertx/pom.xml]: :+1:

File: example/src/main/resources/application.yml [hunk 22]
```diff
@@ -6,7 +6,7 @@ hono:
   client:
     host: localhost
     hostnameVerificationRequired: false
-    trustStorePath: ../config/demo-certs/certs/trusted-certs.pem
+    #trustStorePath: ../config/demo-certs/certs/trusted-certs.pem
```
Comment (sophokles73) [inline example/src/main/resources/application.yml]: Any particular reason for commenting this out?

Comment (ppatierno) [inline example/src/main/resources/application.yml]: No sorry ! My mistake while trying some things ... I'll fix ... now !

File: application/src/main/resources/application-default.yml [hunk 23]
```diff
@@ -0,0 +1,12 @@
+spring:
+  profiles: default
```
Comment (sophokles73) [inline application/src/main/resources/application-default.yml]: I don't think this is necessary since the applicable profile is already selected by means of loading this file ...

File: dispatchrouter/pom.xml [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+        <plugins>
+          <plugin>
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <configuration>
+              <verbose>true</verbose>
+              <images>
+                <image>
+                  <build>
+                    <from>gordons/qpid-dispatch:${qpid-dispatch.version}</from>
```
Comment (sophokles73) [inline dispatchrouter/pom.xml]: I was wondering whether we indeed want to continue to build on top of Gordon's image or whether we want to build the image ourselves from scratch (based on his Dockerfiles). WDYT?

Comment (ppatierno) [inline dispatchrouter/pom.xml]: Yes good idea .... I agree, it's better to be independent. I'll do that ;)

Comment (sophokles73) [inline dispatchrouter/pom.xml]: It's not that I do not trust Gordon with it, but in the past we triggered him whenever a new version of Dispatch Router was released and I think this way we can be, as you said, more independent ...
If you don't mind, would you please also add the `cyrus-sasl-sql` package to the image so that we can also authenticate against a MySQL or Postgresql database?

File: site/content/component/dispatch-router.md [hunk 25]
```diff
@@ -9,18 +9,27 @@ The north bound API is used by applications to consume telemetry data and events
...
+| `/etc/hono/qdrouter/qdrouterd.json`  | The Dispatch Router configuration file. |
+| `/etc/hono/qdrouter/qdrouter-sasl.conf` | The configuration file for the [Cyrus SASL Library](http://www.cyrusimap.org/sasl/getting_started.html) used by Dispatch Router for authenticating clients. This configuration file can be adapted to e.g. configure LDAP or a database for verifying credentials.
+| `/etc/hono/qdrouter/qdrouterd.sasldb` | The Berkley DB file used by Cyrus SASL that contains the example users which are supported by the Dispatch Router.
+
+The file paths are related to the Docker images where such configuration files are needed.
 
 ## Run as a Docker Container
 
-The Dispatch Router can be run as a Docker container from the command line. The following commands first start the volume container containing the [Default Configuration]({{< ref "default-config.md" >}}) and then start the Dispatch Router container mounting the configuration volume in order to be able to read the configuration files from it.
+The Dispatch Router can be run as a Docker container from the command line. The following command starts the Dispatch Router container using the provided configuration files.
```
Comment (sophokles73) [inline site/content/component/dispatch-router.md]: what about "...using the default configuration files included in the Dispatch Router image" ?

File: site/content/component/hono-server.md [hunk 26]
```diff
@@ -36,19 +36,24 @@ The following table provides an overview of the configuration options the adapte
...
+| File                                     | Description                                                      |
+| :--------------------------------------- | :--------------------------------------------------------------- |
+| `/etc/hono/permissions.json`          | Example access control lists for Hono's endpoint resources.      |
+
 ## Run as a Docker Container
 
 When running the Hono server as a Docker container, the preferred way of configuration is to pass environment variables to the container during startup using Docker's `-e` or `--env` command line option.
 
-The following commands first start the volume container containing the [Default Configuration]({{< ref "default-config.md" >}}) and then start the Hono server container mounting the exposed volume in order to be able to read the configuration files from `/etc/hono`.
+The following command starts the Hono server container using the configuration files from `/etc/hono`.
```
Comment (sophokles73) [inline site/content/component/hono-server.md]: "... using the configuration files included in the image under path /etc/hono"?

File: site/content/component/mqtt-adapter.md [hunk 27]
```diff
@@ -30,15 +30,14 @@ The options only need to be set if the default value does not match your environ
 
 When running the adapter as a Docker container, the preferred way of configuration is to pass environment variables to the container during startup using Docker's `-e` or `--env` command line option.
 
-The following commands first start the volume container containing the [Default Configuration]({{< ref "default-config.md" >}}) and then start the MQTT adapter container mounting the exposed volume in order to be able to read the trusted certificates from `/etc/hono/certs`.
+The following command starts the MQTT adapter container using the trusted certificates from `/etc/hono/certs`.
```
Comment (sophokles73) [inline site/content/component/mqtt-adapter.md]: "...trusted certificates included in the image under ..." you get the idea ;-)

File: site/content/component/rest-adapter.md [hunk 28]
```diff
@@ -30,14 +30,13 @@ The options only need to be set if the default value does not match your environ
 
 When running the adapter as a Docker container, the preferred way of configuration is to pass environment variables to the container during startup using Docker's `-e` or `--env` command line option.
 
-The following commands first start the volume container containing the [Default Configuration]({{< ref "default-config.md" >}}) and then start the REST adapter container mounting the exposed volume in order to be able to read the trusted certificates from `/etc/hono/certs`.
+The following command starts the REST adapter container using the trusted certificates from `/etc/hono/certs`.
```
Comment (sophokles73) [inline site/content/component/rest-adapter.md]: and again

File: site/content/user-guide/secure_communication.md [hunk 29]
```diff
@@ -18,13 +18,24 @@ Within a Hono installation the following communication channels can be secured w
...
+| File                                     | Description                                                      |
+| :--------------------------------------- | :--------------------------------------------------------------- |
+| `/etc/hono/certs/qdrouter-key.pem`    | An example private key to use for enabling TLS on the Dispatch Router. |
+| `/etc/hono/certs/qdrouter-cert.pem`   | A certificate containing the public key corresponding to the example private key. |
+| `/etc/hono/certs/trusted-certs.pem`   | A list of the CA certificates used to sign the example keys and certificates. |
+| `/etc/hono/certs/honoKeyStore.p12`    | A key store containing an example private key and matching certificate for enabling TLS on the Hono server. |
+| `/etc/hono/certs/restKeyStore.p12`    | A key store containing an example private key and matching certificate for enabling TLS on the REST adapter. |
+| `/etc/hono/certs/mqttKeyStore.p12`    | A key store containing an example private key and matching certificate for enabling TLS on the MQTT adapter. |
+
+The file paths are related to the Docker images where such certificates and keys are needed for enabling TLS.
```
Comment (sophokles73) [inline site/content/user-guide/secure_communication.md]: I think we should explicitly explain, which files are included in which images.

File: site/content/user-guide/secure_communication.md [hunk 30]
```diff
@@ -18,13 +18,24 @@ Within a Hono installation the following communication channels can be secured w
...
+
 ## Dispatch Router
 
 The dispatch router by default tries to read its configuration from `/etc/qpid-dispatch/qdrouterd.conf`. The default configuration included with the Dispatch Router Docker image can be replaced by means of *mounting a volume* during startup of the Docker image. Please refer to the [Docker documentation for details regarding how to create and use volumes](https://docs.docker.com/engine/tutorials/dockervolumes/).
 
-As part of building the `config` module the `eclipsehono/qpid-default-config` Docker image is created. It exposes configuration files from the `config/qpid` folder as a Docker volume with `/etc/qpid-dispatch` as the volume path. The configuration uses the demo keys and certificates from `config/demo-certs/certs` to enable TLS.
-
-The `eclipsehono/qpid-default-config` volume image is also used by the `example` module to read the Dispatch Router configuration from. Please refer to the readme.md file in the `example` folder for more information.
+Such default configuration uses the demo keys and certificates from `config/demo-certs/certs` to enable TLS mapped to the `/etc/hono/certs` folder in the Docker image.
```
Comment (sophokles73) [inline site/content/user-guide/secure_communication.md]: I do not understand the meaning of this sentence. Can you rephrase it?

File: adapters/mqtt-vertx/pom.xml [hunk 31]
```diff
@@ -146,6 +146,7 @@
                             <outputDirectory>etc/hono/certs</outputDirectory>
                             <includes>
                               <include>trusted-certs.pem</include>
+                              <include>mqttKeyStore.p12</include>
```
Comment (sophokles73) [inline adapters/mqtt-vertx/pom.xml]: +1 makes sense once we want to enable TLS on the adapter

File: site/content/component/mqtt-adapter.md [hunk 32]
```diff
@@ -30,7 +30,7 @@ The options only need to be set if the default value does not match your environ
 
 When running the adapter as a Docker container, the preferred way of configuration is to pass environment variables to the container during startup using Docker's `-e` or `--env` command line option.
 
-The following command starts the MQTT adapter container using the trusted certificates from `/etc/hono/certs`.
+The following command starts the MQTT adapter container using the trusted certificates included in the image under under path `/etc/hono/certs`.
```
Comment (sophokles73) [inline site/content/component/mqtt-adapter.md]: typo: *under under*

File: site/content/component/rest-adapter.md [hunk 33]
```diff
@@ -30,7 +30,7 @@ The options only need to be set if the default value does not match your environ
 
 When running the adapter as a Docker container, the preferred way of configuration is to pass environment variables to the container during startup using Docker's `-e` or `--env` command line option.
 
-The following command starts the REST adapter container using the trusted certificates from `/etc/hono/certs`.
+The following command starts the REST adapter container using the trusted certificates included in the image under under path `/etc/hono/certs`.
```
Comment (sophokles73) [inline site/content/component/rest-adapter.md]: *under under*

File: site/content/user-guide/secure_communication.md [hunk 34]
```diff
@@ -18,24 +18,19 @@ Within a Hono installation the following communication channels can be secured w
 2. Hono server to Dispatch Router - The Hono server connects to the Dispatch Router in order to forward telemetry data and commands hence and forth between downstream components (client applications) and devices. This (internal) connection can be secured by configuring the Dispatch Router and Hono Server for TLS.
 3. Protocol Adapter to Hono server - A protocol adapter connects to Hono to e.g. forward telemetry data received from devices to downstream consumers. This (internal) connection can be secured by configuring the protocol adapter and Hono server for TLS.
 
-The certificates and keys for all the Hono components provided by the `config` module are the following.
+The certificates and keys for all the Hono components are provided by the `config` module in the `config/demo-certs/certs` folder.
```
Comment (sophokles73) [inline site/content/user-guide/secure_communication.md]: Is this still true, i.e. didn't we want to rename the `config` module to `demo-certs` or something? Or will this be done in a separate PR (which I have no problem with at all)?

Comment (ppatierno) [inline site/content/user-guide/secure_communication.md]: It's better to do it know because it's part of this big refactoring ... I'm going to do it :)

Comment (ppatierno) [inline site/content/user-guide/secure_communication.md]: Done !

File: server/pom.xml [hunk 35]
```diff
@@ -63,7 +63,7 @@
     </dependency>
     <dependency>
      <groupId>org.eclipse.hono</groupId>
-     <artifactId>hono-config</artifactId>
+     <artifactId>hono-demo-certs</artifactId>
```
Comment (sophokles73) [inline server/pom.xml]: Not sure if we need this anymore. In the past we included the hono-config.jar in order to also have a copy of permissions.json and application.yml etc on the classpath when running tests. We may need to create a permissions.json file in server/src/test/resources for that purpose now ... can you check?

Comment (ppatierno) [inline server/pom.xml]: You are right ... I needed to create a permissions.json under server/src/test/resources in order to have tests working. I removed the dependency and all works still fine ;)

File: pom.xml [hunk 36]
```diff
@@ -135,13 +135,14 @@
...
     <module>bom</module>
-    <module>config</module>
+    <module>demo-certs</module>
     <module>core</module>
     <module>client</module>
     <module>example</module>
     <module>legal</module>
     <module>server</module>
     <module>tests</module>
+    <module>dispatchrouter</module>
```
Comment (sophokles73) [inline pom.xml]: You may have noted that the modules have been ordered alphabetically so far ... ;-)

File: tests/pom.xml [hunk 37]
```diff
@@ -223,8 +224,11 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
               <images>
                 <!-- ##### ActiveMQ Broker ##### -->
                 <image>
+                  <name>eclipsehono/hono-test-activemq</name>
                   <alias>activemq</alias>
-                  <name>rmohr/activemq:${activemq.version}</name>
+                  <build>
+                    <from>rmohr/activemq:${activemq.version}</from>
+                  </build>
```
Comment (sophokles73) [inline tests/pom.xml]: why do you need to build a new image on top of the existing one?

Comment (ppatierno) [inline tests/pom.xml]: why chat on IRC about this. I removed the useless image building but inverting the position of hono server and router/activemq in the pom.xml file. It's a bug in the docker maven plugin (I had a chat with the fabric8 team about that)

**Comment by vijaykumark** [comment]
Where can I find documentation about running Hono on Openshift?

**Comment by ppatierno** [comment]
The documentation is on the way and will be published soon on the web site. Thanks for your interest !
